src/suit.c

```diff
@@ -393,6 +393,36 @@ int suit_processor_get_manifest_metadata(const uint8_t *envelope_str, size_t env
 		if (manifest_component_id != NULL) {
 			*manifest_component_id = decoder_state->decoded_manifest->manifest_component_id;
 		}
+
+		if ((version != NULL) && (version_len != NULL)) {
+			struct SUIT_Condition_Version_Comparison_Value cbor_version;
+			size_t cbor_version_len;
+
+			/* Initialize returned memory with zero. */
+			memset(version, 0, sizeof(*version) * (*version_len));
+
+			if (decoder_state->decoded_manifest->current_version.len > 0) {
+				ret = cbor_decode_SUIT_Condition_Version_Comparison_Value(
+					decoder_state->decoded_manifest->current_version.value,
+					decoder_state->decoded_manifest->current_version.len,
+					&cbor_version,
+					&cbor_version_len);
+				if ((ret != ZCBOR_SUCCESS) || (cbor_version_len != decoder_state->decoded_manifest->current_version.len)) {
+					ret = SUIT_ERR_DECODING;
+				} if (cbor_version.SUIT_Condition_Version_Comparison_Value_int_count > *version_len) {
+					ret = SUIT_ERR_DECODING;
+				} else {
+					*version_len = cbor_version.SUIT_Condition_Version_Comparison_Value_int_count;
+					for (size_t i = 0; i < *version_len; i++) {
+						version[i] = cbor_version.SUIT_Condition_Version_Comparison_Value_int[i];
+					}
+				}
+			} else {
+				*version_len = 0;
+			}
+		} else if ((version != NULL) || (version_len != NULL)) {
```

Comment:
The idea is that you can selectively ask for some of the metadata. The assumption is that if you are not interested in version, you should not pass `version` and `version_len`. If you want to read them, you must provide both.

Implementing half-queries might be tricky - the (`version == NULL`, `version_len != NULL`) may be interpreted as readout of the expected `version` buffer length and the (`version != NULL`, `version_len == NULL`) is simply invalid.

I thought that it is easier to avoid providing an additional case if we do not see a real use case for it.

The logic for (`digest`, `alg`) pair is the same.